src/js/components/Image.js

```diff
@@ -0,0 +1,69 @@
+import classNames from 'classnames/dedupe';
+import React from 'react';
+import ReactDOM from 'react-dom';
+
+import Util from '../utils/Util';
+
+const METHODS_TO_BIND = ['onImageError'];
+
+class Image extends React.Component {
+  constructor() {
+    super(...arguments);
+
+    this.state = {
+      imageErrorCount: 0
+    };
+
+    METHODS_TO_BIND.forEach((method) => {
+      this[method] = this[method].bind(this);
+    });
+  }
+
+  componentWillReceiveProps(nextProps) {
+    // Try again if new urls were provided
```

Comment:
I figured it didn't matter too much. The only reason why this would be called is if you change the probs, and I believe it easier to not worry about what got changed and start over, which is less error prone. There are no performance gains to be had from this.